File: leetcode_py/Valid Sudoku.py

class Solution:
    # @param board, a 9x9 2D array
    # @return a boolean
    def isValidSudoku(self, board):
        row=[[False for i in range(9)] for j in range(9)]
        col=[[False for i in range(9)] for j in range(9)]
        small=[[False for i in range(9)] for j in range(9)]
        for i in range(9):
            for j in range(9):
                if board[i][j]!='.':
                    num=ord(board[i][j])-ord('1')
                    k=(i/3)*3+(j/3)
                    if row[i][num] or col[j][num] or small[k][num]:
                        return False
                    else:
                        row[i][num]=True
                        col[j][num]=True
                        small[k][num]=True
        return True

File: leetcode_js/Valid Sudoku.js
/**
 * @param {character[][]} board
 * @return {boolean}
 */
var isValidSudoku = function(board) {
    var row = [], col = [], x = [];
    for(var i = 0; i < 9; i++){
    	row[i] = Array(board[0].length);
    	col[i] = Array(board[0].length);
    	x[i] = Array(board[0].length);
    	for(var j = 0; j < 9; j++){
    		row[i][j] = false;
    		col[i][j] = false;
    		x[i][j] = false;
    	}
    }
    for(var i = 0; i < 9; i++){
    	for(var j = 0; j < 9; j++){
    		var num = board[i][j];
    		if(num != '.'){
    			var k = Math.floor(i/3)*3 + Math.floor(j/3);
    			if(row[i][num] || col[j][num] 
    				|| x[k][num]){
    				return false;
    			}
    			row[i][num] = col[j][num] = x[k][num];
    		}
    	}
    }
    return true;
};